pkg/workload/workload.go

```diff
@@ -239,6 +240,9 @@ func UpdateStatusIfChanged(ctx context.Context,
 }
 
 func InCondition(w *kueue.Workload, condition string) bool {
-	i := FindConditionIndex(&w.Status, condition)
-	return i != -1 && w.Status.Conditions[i].Status == metav1.ConditionTrue
+	return InConditionWithStatus(w, condition, metav1.ConditionTrue)
+}
+
+func InConditionWithStatus(w *kueue.Workload, condition string, status metav1.ConditionStatus) bool {
```

Comment:
Don't add this function.

As a follow up: remove `InCondition`

Reply:
https://github.com/kubernetes-sigs/kueue/pull/474